### PR TITLE
Command-line UX overhaul

### LINKED ISSUE #8560: Tracking issue for this set of "terminal revamp" PRs
The "final" set of PRs for this cycle of the terminal revamp need to be reviewed and merged in one release, because merging them separately isn't viable.

This issue tracks their individual submission and approval. 

Once each is approved individually, then we can do a final review and merge.

- [x] #8318: nginx,apache: CLI logging changes
- [x] #8513: cli: redesign output of new certificate reporting
- [x] #8514: cli: redesign output of failed authz reporting
- [x] #8579: cli: dont use IReporter for EFF donation prompt
- [x] #8636: cli: redesign output when cert installation failed
- [x] #8622: renew: don't restart the installer for dry-runs
- [x] #8664: log: print "advice" when exiting with an error
- [x] #8765: cli: reduce default logging verbosity #2

### PR DESCRIPTION
**Pull request by ohemorange**
Fixes #8560.

This PR is a combination of the branches mentioned in #8560, along with any minor fixes we noticed during the merging process.

### CODE REVIEW

File: certbot/CHANGELOG.md [hunk 1]
```diff
@@ -111,6 +116,7 @@ More details about these changes can be found on our GitHub repo.
 * Fixed the apache component on openSUSE Tumbleweed which no longer provides
   an apache2ctl symlink and uses apachectl instead.
 * Fixed a typo in `certbot/crypto_util.py` causing an error upon attempting `secp521r1` key generation
+* Installers (e.g. nginx, Apache) were being restarted unnecessarily after dry-run renewals.
```
Comment (ohemorange) [inline certbot/CHANGELOG.md:119]: oops, this should be in the current release

File: certbot/certbot/_internal/main.py [hunk 2]
```diff
@@ -498,29 +490,70 @@ def _report_new_cert(config, cert_path, fullchain_path, key_path=None):
...
+    """ --csr variant of _report_new_cert.
+
+    Until --csr is overhauled (#8332) this is transitional function to report the creation
+    of a new certificate using --csr.
+    TODO: remove this function and just call _report_new_cert when --csr is overhauled.
+
+    :param config: Configuration object
+    :type config: interfaces.IConfig
+
+    :param cert_path: path to cert.pem
```
Comment (ohemorange) [inline certbot/certbot/_internal/main.py:529]:
```suggestion
    :param str cert_path: path to cert.pem
```

File: certbot/certbot/_internal/main.py [hunk 3]
```diff
@@ -498,29 +490,70 @@ def _report_new_cert(config, cert_path, fullchain_path, key_path=None):
...
+    :type cert_path: str
```
Comment (ohemorange) [inline certbot/certbot/_internal/main.py:530]:
```suggestion
```

File: certbot/certbot/_internal/main.py (hunk 2 above)
Comment (ohemorange) [reply certbot/certbot/_internal/main.py:529]: We only need the fancy two-line version when it's a custom object. This applies to the two below as well.

File: certbot/tests/main_test.py [hunk 4]
```diff
@@ -6,6 +6,7 @@
 import io
 import itertools
 import json
+from operator import is_
```
Comment (ohemorange) [inline certbot/tests/main_test.py:9]: Do we actually use this? Not sure why lint isn't complaining.

**Review by ohemorange — COMMENTED** [review]
Couldn't find any weird inclusions or exclusions! A couple minor nits, but otherwise, code looks good. Also, since I opened this PR, I can't approve or request changes.

**Comment by alexzorin** [comment]
Before/After @ f104d1f9bf53eb2e625f03889489906dc87bd99c: https://docs.google.com/document/d/1-u5LfxDPyXDDeIXnZ5JG92eUEK-iugJ0lIYawYheP70/edit?usp=sharing

**Review by alexzorin — APPROVED** [review]

**Pull request merged**